ts/Extensions/Data.ts

```diff
@@ -1429,95 +1453,34 @@ class Data {
             return format;
         }
 
-        /**
-         * @todo
-         * Figure out the best axis types for the data
-         * - If the first column is a number, we're good
-         * - If the first column is a date, set to date/time
-         * - If the first column is a string, set to categories
-         * @private
-         */
-        function deduceAxisTypes(): void {
-
-        }
-
-        if (csv && options.beforeParse) {
-            csv = options.beforeParse.call(this, csv);
-        }
+        let csvStore;
 
         if (csv) {
+            csvStore = this.dataStore = new CSVStore(
+                new DataTable(),
+                {
+                    csv: csv,
+                    csvURL: options.csvURL,
+                    enablePolling: options.enablePolling,
+                    dataRefreshRate: options.dataRefreshRate
+                },
+                new CSVParser({
+                    csv: csv,
+                    startRow: options.startRow,
+                    endRow: options.endRow,
+                    startColumn: options.startColumn,
+                    endColumn: options.endColumn,
+                    firstRowAsNames: options.firstRowAsNames,
+                    switchRowsAndColumns: options.switchRowsAndColumns,
+                    decimalPoint: options.decimalPoint,
+                    itemDelimiter: options.itemDelimiter,
+                    lineDelimiter: options.lineDelimiter
```

Comment:
Why the repetition? Can we just pass on either `options` or `merge(options)`?

Reply:
As the options are a combination for table and parser constructor, this assignment is nearly as good as it can get. Your suggestions would flood each constructor with options for the other one. Or is this what your meant?

Reply:
It is not possible because of overlapping options. :(